### ISSUE TITLE
Normalize tabs for json formatter the same way we do for gcc

### ISSUE THREAD

**Issue by pwielgolaski** [koalaman/shellcheck#1048]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
	 [[ $n != 0 ]]
```

#### Here's what shellcheck currently says:
```
➜  sample shellcheck -f json issue.bash | jq
[
  {
    "file": "issue.bash",
    "line": 2,
    "endLine": 2,
    "column": 13,
    "endColumn": 13,
    "level": "warning",
    "code": 2154,
    "message": "n is referenced but not assigned."
  }
]
➜  sample shellcheck -f gcc issue.bash      
issue.bash:2:6: warning: n is referenced but not assigned. [SC2154]

```

#### Here's what I wanted or expected to see:

In gcc format we normalize tabs, could we do the same for json formatter, so we have the same column number in both?

**Comment by koalaman** [comment]
The formatters follow the convention of the tools they mimic. `gcc` uses a tab size of 1, `checkstyle` uses a tab size of 8. Changing those would break compatibility.

There's no particular reason why the shellcheck JSON format couldn't have used the `gcc` style instead of `checkstyle` and `tty`, but it didn't, and I don't see any value in changing it now since that would break the integrations that rely on it (like Syntastic).

The tab size used by each formatter is documented: 
https://github.com/koalaman/shellcheck/wiki/Integration

**Comment by pwielgolaski** [comment]
Make sense, I did not see it being documented.
It would make easier my integration with intellij, but I can handle it in code.
Btw. I implemented integration plugin.
https://github.com/pwielgolaski/shellcheck-plugin waiting for approval to get it available

**Comment by blueyed** [comment]
Just for information: typically linters use the byte/character offset, and do not handle tab in a special way (therefore a tab in gcc has an offset of 1 I assume).

Regarding compatibility: Syntastic uses `-fgcc`, not the json formatter.

It is not trivial to adjust this in Vim, where you would have to look at the actual line and adjust the column offset based on the tabs before the reported column.
